api/v1/controllers/AuthController.js

```diff
@@ -70,8 +80,85 @@ function refreshToken (req, res, next) {
 		});
 }
 
+function requestPasswordReset (req, res, next) {
+	var userEmail = req.query.email;
+
+	UserService
+		.findUserByEmail(userEmail)
+		.then(function (user){
+			return AuthService.generateToken(user, scopes.AUTH);
+		})
+		.then(function (tokenVal){
+			//TODO: Determine exact url route for reset page
+			tokenURL = (config.isDevelopment ? "localhost:8080/" : "hackillinois.org/") + "v1/auth/resetpage?token=" + tokenVal;
+			var substitutions = {'resetURL': tokenURL};
+			return MailService.send(userEmail, mail.templates.passwordReset, substitutions);
+		})
+		.then(function(){
+			res.body = {};
+			next();
+			return null;
+		})
+		.catch(function (error){
+			next(error);
+			return null;
+		});
+}
+
+function passwordReset(req, res, next) {
+	// MARK - Should be able to remove this and let middleware handle the requirements
+	/* 
+	if (req.body.password.length < MINIMUM_PASSWORD_LENGTH)
+	{
+		var message = 'Password must be at least ' + MINIMUM_PASSWORD_LENGTH + ' characters long';
+		next(new errors.InvalidParameterError(message, 'password'));
+		return;
+	}
+	*/
+
+	AuthService
+		.findTokenByValue(req.body.token)
+		.then(function (token) {
+			var tokenExpire = Date.parse(token.get('created')) + config.tokenExpiration.AUTH;
+			if (tokenExpire < Date.now())
+			{
+				// Invalid token (expired)
+				token.destroy().then(function () {
+					var message = 'Cannot reset password: current token is expired.' +
+						    ' Please request a new token via API';
```

Comment:
Can we change this message to just say "The provided token has expired"? It seems like that's more concise (the client already knows where to request a token and that the password isn't reset if they get a non-200 error).